### PR TITLE
Add scope option to oauth2 refresh function

### LINKED ISSUE #18: Support scopes when refreshing token
According to the OAuth2 specification a call to the token endpoint with the `refresh_token` grant type can optionally accept scopes. Oslos OAuth2 client implementation does not currently support sending scopes in its `refreshAccessToken` function.

Relevant spec: https://datatracker.ietf.org/doc/html/rfc6749#section-6

### PR DESCRIPTION
**Pull request by LinusOP**
This adds the option to send scopes in the `refreshAccessToken` function of the OAuth2 client.

Fixes #18

### CODE REVIEW

**Review by pilcrowonpaper — COMMENTED** [review]
Can you update the docs as well?

**Comment by LinusOP** [comment]
> Can you update the docs as well?

Ofc, to be honest I did this at about 5 in the morning and completely forgot 😅
Will sort now

**Comment by LinusOP** [comment]
Docs have been added, think they should cover it

File: docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md [hunk 1]
```diff
@@ -30,6 +31,7 @@ function refreshAccessToken<_TokenResponseBody extends TokenResponseBody>(
 - `options`
   - `credentials`: Client password or secret for authenticated requests
   - `authenticateWith` (default: `"http_basic_auth"`): How the credentials should be sent
+  - `scopes`: Optional scopes to pass to the token endpoint when refreshing. Certain providers require this.
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md]: Mostly to be consistent with other pages

```suggestion
  - `scopes`
```

File: docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md [hunk 2]
```diff
@@ -50,7 +52,8 @@ interface ResponseBody extends TokenResponseBody {
 try {
 	const tokens = await $$oauth2Client.refreshAccessToken<ResponseBody>(code, {
 		credentials: clientSecret,
-		authenticateWith: "request_body" // send client secret inside body
+		authenticateWith: "request_body", // send client secret inside body
+		scopes: [] // optional scopes, required by some providers
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md]:
```suggestion
		scopes: []
```

File: docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md [hunk 3]
```diff
@@ -50,7 +52,8 @@ interface ResponseBody extends TokenResponseBody {
 try {
 	const tokens = await $$oauth2Client.refreshAccessToken<ResponseBody>(code, {
 		credentials: clientSecret,
-		authenticateWith: "request_body" // send client secret inside body
+		authenticateWith: "request_body", // send client secret inside body
+		scopes: []
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md]:
```suggestion
```

File: docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md [hunk 4]
```diff
@@ -50,7 +52,7 @@ interface ResponseBody extends TokenResponseBody {
 try {
 	const tokens = await $$oauth2Client.refreshAccessToken<ResponseBody>(code, {
 		credentials: clientSecret,
-		authenticateWith: "request_body" // send client secret inside body
+		authenticateWith: "request_body", // send client secret inside body
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/OAuth2Client/refreshAccessToken.md]:
```suggestion
		authenticateWith: "request_body" // send client secret inside body
```

File: docs/pages/reference/oauth2/index.md [hunk 5]
```diff
@@ -115,7 +115,8 @@ try {
 		refreshToken: string;
 	}>(code, {
 		credentials: clientSecret,
-		authenticateWith: "request_body"
+		authenticateWith: "request_body",
+		scopes: []
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/index.md]:
```suggestion
```

File: docs/pages/reference/oauth2/index.md [hunk 6]
```diff
@@ -115,7 +115,7 @@ try {
 		refreshToken: string;
 	}>(code, {
 		credentials: clientSecret,
-		authenticateWith: "request_body"
+		authenticateWith: "request_body",
```
Comment (pilcrowonpaper) [inline docs/pages/reference/oauth2/index.md]:
```suggestion
		authenticateWith: "request_body"
```

**Review by pilcrowonpaper — APPROVED** [review]

**Comment by pilcrowonpaper** [comment]
Thanks!

**Pull request merged**